tests/test_fork_provider.py

```diff
@@ -146,13 +149,20 @@ def test_transaction_contract_as_sender(
 
     with pytest.raises(ContractLogicError, match="!authorized"):
         # Task failed successfully
-        mainnet_fork_contract_instance.setNumber(10, sender=mainnet_fork_contract_instance)
+        # NOTE: For some reason, this only fails when for estimate gas. Otherwise, the status
```

Comment:
```suggestion
        # NOTE: For some reason, this only fails for estimate gas. Otherwise, the status
```

also can we make sure this is still True? I know this was copied from ape-foundry, i am not sure if it is actually applicable here.

Reply:
Double checked and the test passed without the need for setting the gas so I'm removing it.